### PR TITLE
Added more tests to increase efficiency of the script

### PR DESCRIPTION
**Pull request by YousafAzabi**
Started extracting functions from comparator.js and put them in separate files (modules).
Generated angle.js, distance.js and data.js. 
Wrote tests to cover more scenarios and increase efficiency.

### CODE REVIEW

File: src/comparator/distance.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
+//module to find distance between first node of two links
+
+const turf = require('@turf/turf');
+
+exports.inRange = (roadOne, roadTwo, range = 1) => {
```
[the remark this answers is not in the document]
Comment (sebastianovide) [reply src/comparator/distance.js]: @YousafAzabi  any ideas on how to reduce it ?

File: src/comparator/distance.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+  // find 1 point in second road
+  const pointTwo = roadTwo.geometry.coordinates[0];
+  // find distance
+  const distance = turf.distance(pointOne, pointTwo);
+  // if longer than the range (in km) return false
+  console.log(distance);
+  if (distance > range) {
+    return false;
+  }
+  return true;
```
[the remark this answers is not in the document]
Comment (sebastianovide) [reply src/comparator/distance.js]: @YousafAzabi  any ideas on how to reduce it ?

File: src/comparator/distance.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (YousafAzabi) [reply src/comparator/distance.js]: I do not know what do they mean by Cognitive Complexity. I have these warning in 5 or 6 files and i want to discuss with you.

File: src/comparator/distance.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (YousafAzabi) [reply src/comparator/distance.js]: I think I can reduce them. I will work on it.

File: src/comparator/distance.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (sebastianovide) [reply src/comparator/distance.js]: cool. Let's brainstorm together.

**Pull request merged**